aiida_common_workflows/workflows/relax/quantum_espresso/extractors.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+"""
+Post-processing extractor functions for the `QuantumEspressoCommonRelaxWorkChain`.
+"""
+
+from .workchain import QuantumEspressoCommonRelaxWorkChain
+
+
+def get_ts_energy(common_relax_workchain: QuantumEspressoCommonRelaxWorkChain) -> float:
+    """
+    Return the TS value of a concluded QuantumEspressoCommonRelaxWorkChain.
+    T the fictitious temperature due to the presence of a smearing and S is
+    the entropy.
+
+    :param common_relax_workchain: Quantum ESPRESSO common relax work chain for which to extract the smearing energy.
+
+    :returns: Smearing energy in eV.
+    """
+    from aiida.common import LinkType
+    from aiida.orm import WorkChainNode
+    from aiida.plugins import WorkflowFactory
+
+    if not isinstance(common_relax_workchain, WorkChainNode):
+        return ValueError('The input is not a workchain (instance of `WorkChainNode`)')
+    if common_relax_workchain.process_class != WorkflowFactory('common_workflows.relax.quantum_espresso'):
+        return ValueError('The input workchain is not a `QuantumEspressoCommonRelaxWorkChain`')
+
+    qe_relax_wc = common_relax_workchain.get_outgoing(link_type=LinkType.CALL_WORK).one().node
+    return -qe_relax_wc.outputs.output_parameters['energy_smearing']
```

Comment:
I take it the minus sign is intentional. Should we add a comment why this is?

Reply:
Added a full explanation in the docstring. 👍

Reply:
Where is this function used though? I don't see it anywhere? Shouldn't this be at the end of the `QuantumEspressoCommonRelaxWorkChain` to attach the smearing contribution as an output node? And if so, should this not really be a calcfunction?

Reply:
It's used in the verification project, since there can be a significant difference in the EOS curves if you use the internal energy E or the free energy E - TS. However, I also [can't find it when looking through the `acwf-verification-scripts` package](https://github.com/search?q=repo%3Aaiidateam%2Facwf-verification-scripts%20get_ts_energy&type=code). Maybe @giovannipizzi or @bosonie can comment?

Reply:
So it's basically a feature for a user that wants to investigate the influence of free vs internal energy. It's used for one of the analyses in the supplementary information of the verification paper, but not used in any other parts of the code. Just there as a utility function.

Maybe it would be better to add this to the `tools` namespace, but then we'd have to do this for all the work chains. So let's leave it here for now and refactor later in case we think this makes sense.